src/RG.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+//------------------------------------------------------------------------------
+// code development settings
+//------------------------------------------------------------------------------
+
+// to turn on Debug for a single file add:
+// #define RG_DEBUG
+// just before the statement:
+// #include "RG.h"
+
+// to turn on Debug globaly, uncomment this line:
+// #define RG_DEBUG
+
+//------------------------------------------------------------------------------
+// debugging definitions
+//------------------------------------------------------------------------------
+
+#undef ASSERT
```

Comment:
As we talked in our conversation, personally I don't like to `undef` something that someone else wrote, and define my own. I think that using `RG_ASSERT` is giving better usage context, and unifies coding abstractions like `RG_ASSERT`, `RG_DEBUG`,  `RG_MALLOC`, `RG_DO_SOMETHING`

Reply:
I also favor `RG_ASSERT` as a name and convention; I feel like it avoids potential problems with namespace collisions.

Reply:
Because this is a compile time macro, I don't see how you can have a namespace collision.
GraphBLAS has the exact same macro but it is compiled separately, otherwise I believe you would see a compiler complaining for re-definition.